### ISSUE TITLE
DisMax with array of queries is broken

### ISSUE THREAD

**Issue by Destroy666x** [ruflin/Elastica#1581]
If you pass an array, which the function seemingly supports based on the exception/condition but not the function name, you get a `query malformed, must start with start_object` error. Moving the logic to separate `addQueries` function would be more logical

**Comment by p365labs** [comment]
hi @Destroy666x 
- which version of Elastica r u using and which of Elasticsearch ?
- can you give us some example in order to have reproduce the same error ?

**Comment by Destroy666x** [comment]
Latest, 6.3.x. But it'll happen with any ES version.

The example is that this https://github.com/ruflin/Elastica/blob/master/lib/Elastica/Query/DisMax.php#L18 and this https://github.com/ruflin/Elastica/blob/master/lib/Elastica/Query/DisMax.php#L27 line suggest that the parameter can be an array of objects extending `AbstractQuery`, but when you pass an array you simply get the mentioned error.

**Comment by p365labs** [comment]
@Destroy666x got it, can u show me a simple array which raises the error ?

**Comment by p365labs** [comment]
@Destroy666x #1584 PR with test proving the array works well. Probably you have tested it with a wrong array.

**Comment by Destroy666x** [comment]
@p365labs Yes, you're right I had some confusion between BoolQuery and this since I used both next to eachother. My bad.
